docs/cookbook/using-docker.md

```diff
@@ -0,0 +1,321 @@
+
+First, make sure your `package-lock.json` file is up to date by running `npm update` on your project repo. Then run:
+
+```bash
+docker compose build
+```
+
+After your container is re-built run:
+
+```bash
+docker compose restart
+```
+
+### Summary
+While in this example, our project is still being hosted locally, any of these commands can be issued on a server that supports Docker once your project is deployed.
+
+Right now, our Dockerized container is limited to a single server. For simple, low-traffic sites this could be fine. However, if we want to scale our site over several servers and add a load balancer like Nginix, we need to add support for cloud storage and a cloud database. Fortunately, Apostrophe makes this relatively easy.
+
+## Using AWS S3 services
+If you aren't hosting your project on a single server, you will need to have a different uploaded asset storage method. Typically this is a service like Amazon Web Services S3 or another similar service. Apostrophe is set up to easily use S3 services by adding environment variables. You can read more in the [documentation](https://v3.docs.apostrophecms.org/reference/modules/uploadfs.html#s3-storage-options). We can take advantage of this in Docker by expanding our `docker-compose.yml` and `.env` files.
+
+### Changing the `docker-compose.yaml` file
+In order to pass the environment variables into our project container we just need to add them inside the `environment:` key. If we are using S3 services at Amazon, we need to add four variables: `APOS_S3_REGION`, `APOS_S3_BUCKET`, `APOS_S3_KEY`, and `APOS_S3_SECRET`. For other S3-type storage solutions, such as [filebase](https://filebase.com/), you will also want to set the `APOS_S3_ENDPOINT` variable. For AWS, your `environment:` section should now look like this:
+
+<AposCodeBlock>
+
+```bash
+…
+    environment:
+      - NODE_ENV
+      - APOS_MONGODB_URI
+      - APOS_CLUSTER_PROCESSES
+      - APOS_S3_REGION
+      - APOS_S3_BUCKET
+      - APOS_S3_KEY
+      - APOS_S3_SECRET
+…
+```
+
+<template v-slot:caption>
+  docker-compose.yaml
+</template>
+
+</AposCodeBlock>
+
+### Changing the `.env` file
+Next, the `.env` file should be modified to contain values for each of the new environment variables. Each will get populated with values specific to your S3 buckets. Again, add the `APOS_S3_ENDPOINT` with value if using a service not hosted by AWS.
+
+<AposCodeBlock>
+
+```sh
+NODE_ENV=production
+APOS_MONGODB_URI=mongodb://db:27018/apostrophe
+APOS_S3_REGION=<your region>
+APOS_S3_BUCKET=<your bucket name>
+APOS_S3_KEY=<account key>
+APOS_S3_SECRET=<account secret>
+```
+
+<template v-slot:caption>
+  .env
+</template>
+
+</AposCodeBlock>
+
+While setting S3 permissions is beyond the scope of this guide, you might have to either change your `@apostrophecms/uploadfs` options for accessing the bucket or make changes to the ACL policies through your AWS dashboard.
```

Comment:
I hate to say it, but... it's not beyond the scope of this guide. It's not going to work unless they do it right, so this will be a frustrating document if we don't cover it. I suggest including the screenshots I provided.

Reply:
Images and text added.